src/components/forms/fields/InputField/InputField.tsx

```diff
@@ -30,8 +30,11 @@ export type InputFieldProps = ComponentProps<'input'> & {
   /** Tags to be displayed inside the input field */
   tags?: undefined | string[],
 
-  /** Callback to remove a specific Tag, passed down to Tag component. */
-  tagRemoveCallback?: (index: number) => void,
+  /** Callback to update the input value. Internally hooks to onChange */
+  onUpdate?: undefined | ((arg0: string) => void),
+
+  /** Callback to update the tags. Internally hooks to onKeyUp */
+  onUpdateTags?: undefined | ((arg0: string[]) => void),
```

Comment:
Instead of `arg0` please give these some meaningful names. For example `value` and `tags`, respectively.